pkg/restful/middleware/middleware_test.go

```diff
@@ -0,0 +1,81 @@
+//go:build test_unit
+
+/*
+Copyright 2017 The Nuclio Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package middleware
+
+import (
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/go-chi/chi/v5"
+	"github.com/nuclio/logger"
+	nucliozap "github.com/nuclio/zap"
+	"github.com/stretchr/testify/suite"
+)
+
+type middlewareTestSuite struct {
+	suite.Suite
+	logger         logger.Logger
+	router         chi.Router
+	testHTTPServer *httptest.Server
+}
+
+func (suite *middlewareTestSuite) SetupTest() {
+	suite.logger, _ = nucliozap.NewNuclioZapTest("test")
+
+	// root router
+	suite.router = chi.NewRouter()
+
+	// set the router as the handler for requests
+	suite.testHTTPServer = httptest.NewServer(suite.router)
+}
+
+func (suite *middlewareTestSuite) TearDownTest() {
+	suite.testHTTPServer.Close()
+}
+
+func (suite *middlewareTestSuite) TestModifyIguazioRequestHeaderPrefix() {
+
+	// create a handler to use as "next" which will verify the request headers
+	nextHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		for header := range r.Header {
+			suite.Require().NotContains(header, "igz")
```

Comment:
assert that `x-igz-something` turned into `x-nuclio-something`

Reply:
The `Modify Labels` function does not add `nuclio` to the labels, it only removes `igz`.
I saw that that the labels that needed changing were `x-igz-nuclio-something` when they arrive.

Still, I will assert the actual expected value.

Reply:
oh, got it. correct. i originally typoed the `x-igz-something`, was suppose to be
`x-igz-nuclio-something` -> `x-nuclio-something`